scripts/proto/lbann/models/resnet.py

```diff
@@ -0,0 +1,536 @@
+import lbann.lbann_proto as lp
+import lbann.lbann_modules as lm
+
+# ==============================================
+# Helper modules
+# ==============================================
+
+class ConvBNRelu(lm.Module):
+    """Convolution -> Batch normalization -> ReLU
+
+    Basic unit for ResNets. Assumes image data in NCHW format.
+
+    """
+
+    def __init__(self, out_channels, kernel_size, stride, padding,
+                 bn_zero_init, bn_stats_aggregation,
+                 relu, name):
+        """Initialize ConvBNRelu module.
+
+        Args:
+            out_channels (int): Number of output channels, i.e. number
+                of convolution filters.
+            kernel_size (int): Size of convolution kernel.
+            stride (int): Convolution stride.
+            padding (int): Convolution padding.
+            bn_zero_init (bool): Zero-initialize batch normalization
+                scale.
+            bn_stats_aggregation (str): Aggregation mode for batch
+                normalization statistics.
+            relu (bool): Apply ReLU activation.
+            name (str): Module name.
+
+        """
+        super().__init__()
+        self.name = name
+        self.instance = 0
+
+        # Initialize convolution
+        self.conv = lm.Convolution2dModule(out_channels, kernel_size,
+                                           stride=stride, padding=padding,
+                                           bias=False,
+                                           name=self.name + '_conv')
+
+        # Initialize batch normalization
+        bn_scale_init = 0.0 if bn_zero_init else 1.0
+        bn_scale = lp.Weights(initializer=lp.ConstantInitializer(value=bn_scale_init),
+                              name=self.name + '_bn_scale')
+        bn_bias = lp.Weights(initializer=lp.ConstantInitializer(value=0.0),
+                             name=self.name + '_bn_bias')
+        self.bn_weights = [bn_scale, bn_bias]
+        self.bn_stats_aggregation = bn_stats_aggregation
+
+        # Initialize ReLU
+        self.relu = relu
+
+    def __call__(self, x):
```

Comment:
We should be consistent about whether we overload `forward` or `__call__`. Both are equivalent, but it's probably prettier for users to overload `forward`.

Reply:
Good catch. Fixed.